packages/pwa-kit-extension-sdk/src/types.ts

```diff
@@ -69,12 +62,11 @@ export type ComponentMap = {
     [key: string]: React.ComponentType<any>
 }
 
-/**
- * This type is used in the deserialize() method of an Application Extension to represent
- * the data from an extension after it has been deserialized on the client.
+/*
+ * Same as react-router's RouteProps, but now supports `componentName` if the route is meant to be serializable
  */
-export type DeserializedExtension = {
-    routes: RouteProps[]
+export type SerializedRouteProps = RouteProps & {
+    componentName?: string
 }
```

Comment:
introduced a new type `SerializedRouteProps` that is returned by the `serialize` method